### PR TITLE
Various JavaScript import fixes

### PR DESCRIPTION
**Pull request by stephenmdangelo**

### CODE REVIEW

**Comment by stephenmdangelo** [comment]
If merged, I believe this supercedes #246.

File: src/engine/jsparser.js [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
-    with(QmlWeb) with ($context) {
-      ${source}
-    }
-    ${jsData.exports.map(sym => `$context.${sym} = ${sym};`).join("")}
-  `)(jsData, $context);
+  /* Set the QmlWeb.executionContext so that any internal calls to Qt.include
+   * will have the proper context */
+  const oldExecutionContext = QmlWeb.executionContext;
+  QmlWeb.executionContext = $context;
+  contextSetter($context);
```
Comment (ChALkeR) [inline src/engine/jsparser.js]: Lint says that `contextSetter` is not defined here.

Comment (stephenmdangelo) [inline src/engine/jsparser.js]: Yes, a bad rebase that I haven't pushed the fix for...

Comment (stephenmdangelo) [inline src/engine/jsparser.js]: Fixed.

**Comment by ChALkeR** [comment]
Uh-oh. I will try to take a closer look at this soon =).

**Comment by ChALkeR** [comment]
It looks like this was incorrectly split into commits, 738509c2 (#320) introduced and checked `this.js`, but it wasn't populated, and now 8b4c5a33ea46d76ea6c1038b2ed25b0cbf218a02 populates it. I didn't notice that while reviewing #320.

File: src/modules/QtQml/Qt.js [hunk 2]
```diff
@@ -105,6 +105,34 @@ const Qt = {
     return new QmlWeb.QSizeF(width, height);
   },
 
+  include(path)
+  {
```
Comment (ChALkeR) [inline src/modules/QtQml/Qt.js]: Nit: `include(path) {`. Not sure why that isn't linted already.

File: src/modules/QtQml/Qt.js [hunk 3]
```diff
@@ -105,6 +105,34 @@ const Qt = {
...
+    const engine = QmlWeb.engine;
+
+    const uri = engine.$resolvePath(path);
+
+    /* Handle recursive includes */
+    if (QmlWeb.executionContext.$qmlJsIncludes === undefined) {
+      QmlWeb.executionContext.$qmlJsIncludes = [];
+    }
+
+    if (QmlWeb.executionContext.$qmlJsIncludes.indexOf(uri) >= 0)
```
Comment (ChALkeR) [inline src/modules/QtQml/Qt.js]: Nit: `if (QmlWeb.executionContext.$qmlJsIncludes.indexOf(uri) >= 0) {` (that should fit). Not sure why that isn't linted already.

**Comment by stephenmdangelo** [comment]
Yeah, #320 erroneously didn't actually use the cache that was introduced at all...

**Comment by ChALkeR** [comment]
Btw, does this fix #270?

File: src/modules/QtQml/Qt.js (hunk 3 above)
Comment (ChALkeR) [inline src/modules/QtQml/Qt.js]: Lint should be fixed in 3e39497.

**Comment by ChALkeR** [comment]
Could you rebase this, please? I believe that the hack in `tests/QMLEngine/qml/importTestInclude.js` won't be needed now.

**Comment by ChALkeR** [comment]
Ok, I fixed the nits and rebased.

**Comment by ChALkeR** [comment]
Landed in cd9e44e...84fcab1, thanks!

**Comment by stephenmdangelo** [comment]
Thanks :)

**Pull request merged**